plugins/pur/core/src/main/java/org/pentaho/di/repository/pur/PurRepository.java

```diff
@@ -2788,19 +3106,30 @@ public RepositoryObject getObjectInformation( ObjectId objectId, RepositoryObjec
     throws KettleException {
     try {
       RepositoryFile repositoryFile;
+      readWriteLock.readLock().lock();
       try {
         repositoryFile = pur.getFileById( objectId.getId() );
       } catch ( Exception e ) {
         // javax.jcr.Session throws exception, if a node with specified ID does not exist
         // see http://jira.pentaho.com/browse/BISERVER-12758
         log.logError( "Error when trying to obtain a file by id: " + objectId.getId(), e );
         return null;
+      } finally {
+        readWriteLock.readLock().unlock();
```

Comment:
The read lock should not be release between this and lower in the same method. The lock should be kept.